src/jit/scopeinfo.cpp

```diff
@@ -154,6 +154,78 @@ CodeGenInterface::siVarLoc::siVarLoc(const LclVarDsc* varDsc, regNumber baseReg,
     }
 }
 
+//------------------------------------------------------------------------
+// An overload for the inequality comparator
+bool CodeGenInterface::siVarLoc::operator!=(const siVarLoc* rhs) const
+{
+    return !this->operator==(rhs);
+}
+
+//------------------------------------------------------------------------
+// An overload for the equality comparator
+bool CodeGenInterface::siVarLoc::operator==(const siVarLoc* rhs) const
+{
+    bool areEquals = vlType == rhs->vlType;
```

Comment:
Early returns tend to work better in such cases. You don't have to go 40 lines below to see that it just returns `areEquals` nor do you need an extra level of indentation due to the `if` below.

Reply:
But that would imply adding more than one return to a function. Are two returns clearer than just one?

Reply:
> But that would imply adding more than one return to a function. Are two returns clearer than just one?

It depends. If function fits on the screen (in ideal world they all should) and early returns makes code more straightforward.

Reply:
In general I do not think that the number of returns alone is meaningful. It matters where the returns are and the overall flow of the function. If you have a large function with complicated flow (nested ifs, loops etc.) and put returns in the middle of the function then, well, you deserve to be locked away and never allowed to write code again.

But that's not the case here, the flow control is trivial, linear. It could have 1000 returns and still be OK (though at that point the function would be unreasonably large anyway).

Reply:
I have broken down the long expression and rewritten it.